packages/target-decisioning-engine/src/artifactProvider.js

```diff
@@ -109,26 +114,38 @@ function ArtifactProvider(config) {
    * @return {Promise<import("../types/DecisioningArtifact").DecisioningArtifact>}
    */
   function deobfuscate(res) {
-    return artifactFormat === ARTIFACT_FORMAT_BINARY
-      ? res
-          .arrayBuffer()
-          .then(buffer => obfuscationProvider.deobfuscate(buffer))
-      : res.json();
+    if (artifactFormat === ARTIFACT_FORMAT_BINARY) {
+      perfTool.timeStart(TIMING_ARTIFACT_DEOBFUSCATE);
+      return res.arrayBuffer().then(buffer => {
+        return obfuscationProvider.deobfuscate(buffer).then(deobfuscated => {
+          perfTool.timeEnd(TIMING_ARTIFACT_DEOBFUSCATE);
```

Comment:
we should probably reduce some of this nesting by extracting methods

Reply:
it was a lot prettier before adding the timings 😢